src/tasks/components/TaskHeaderCard.tsx

```diff
@@ -0,0 +1,174 @@
+import {withRouter, RouteComponentProps} from 'react-router-dom'
+import Moment from 'react-moment'
+
+// Components
+import {
+  SlideToggle,
+  ComponentSize,
+  ResourceCard,
+  InputLabel,
+  FlexBox,
+  AlignItems,
+  FlexDirection,
+  JustifyContent,
+} from '@influxdata/clockface'
+
+// Actions
+import {
+  addTaskLabel,
+  deleteTaskLabel,
+  runTask,
+  getRuns,
+} from 'src/tasks/actions/thunks'
+
+// Types
+import {ComponentColor, Button} from '@influxdata/clockface'
+import {Task, AppState} from 'src/types'
+
+interface PassedProps {
+  task: Task
+  onActivate: (task: Task) => void
+  onEditTask: () => void
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = PassedProps & ReduxProps
+
+export class TaskHeaderCard extends PureComponent<
+  Props & RouteComponentProps<{orgID: string; id: string}>
+> {
+  public render() {
+    const {task} = this.props
+
+    if (!task) {
+      return null
+    }
+
+    return (
+      <ResourceCard
+        testID="task-card"
+        disabled={!this.isTaskActive}
+        alignItems={AlignItems.Center}
+        margin={ComponentSize.Large}
+        direction={FlexDirection.Row}
+        justifyContent={JustifyContent.SpaceBetween}
+      >
+        <FlexBox
+          alignItems={AlignItems.FlexStart}
+          direction={FlexDirection.Column}
+          margin={ComponentSize.Large}
+        >
+          <ResourceCard.Name name={task.name} testID="task-card--name" />
+          <ResourceCard.Meta>
+            {this.activeToggle}
+            <>Created at: {task.createdAt}</>
+            <>Created by: {task.name}</>
+            <>
+              Last Used:{' '}
+              <Moment fromNow ago>
+                {task.latestCompleted}
+              </Moment>{' '}
+              ago
+            </>
+            <>{task.org}</>
+          </ResourceCard.Meta>
+        </FlexBox>
+
+        <FlexBox
+          justifyContent={JustifyContent.FlexEnd}
+          direction={FlexDirection.Row}
+        >
+          <Button
+            onClick={this.handleRunTask}
+            text="Run Task"
+            className="task-header-card-btn"
+          />
+          <Button
+            onClick={this.handleEditTask}
+            text="Edit Task"
+            color={ComponentColor.Primary}
+          />
+        </FlexBox>
+      </ResourceCard>
+    )
+  }
+
+  private get activeToggle(): JSX.Element {
+    const labelText = this.isTaskActive ? 'Active' : 'Inactive'
+    return (
+      <FlexBox margin={ComponentSize.Small}>
+        <SlideToggle
+          active={this.isTaskActive}
+          size={ComponentSize.ExtraSmall}
+          onChange={this.changeToggle}
+          testID="task-card--slide-toggle"
+        />
+        <InputLabel active={this.isTaskActive}>{labelText}</InputLabel>
+      </FlexBox>
+    )
+  }
+
+  private handleRunTask = () => {
+    const {onRunTask, match, getRuns} = this.props
+    try {
+      onRunTask(match.params.id)
+      getRuns(match.params.id)
```

Comment:
is it worth setting the `getRuns` in a `finally` here, so that regardless of whether there's an error or not, we'd want to retrieve the full list of runs?